### PR TITLE
Fix missing data in aggregates with bz1

### PR DESCRIPTION
**Pull request by pauldix**
this is ready to go

### CODE REVIEW

File: cmd/influxd/run/server_test.go [hunk 1]
```diff
@@ -994,6 +994,66 @@ func TestServer_Query_Count(t *testing.T) {
...
+	writes := []string{
+		"cpu,host=server1 value=1.0 10000000000",
+		"cpu,host=server1 value=2.0 20000000000",
+		"cpu,host=server1 value=3.0 30000000000",
+	}
+	test.write = strings.Join(writes, "\n")
+
+	test.addQueries([]*Query{
+		&Query{
+			name:    "selecting count(value) should succeed",
```
Comment (otoolep) [inline cmd/influxd/run/server_test.go]: Maybe add a second test here that does the `count` without any `GROUP BY`, and enhance names of tests to call out difference.

**Comment by jwilder** [comment]
Tested locally and this fixes #3781 for me. :+1:

**Comment by Jhors2** [comment]
Any chance you can trigger a new RC build with this or will we have to wait until this evening?  This bug was a show stopper for me to upgrade.

File: tsdb/engine/bz1/bz1.go [hunk 2]
```diff
@@ -613,7 +619,12 @@ type Cursor struct {
 // Seek moves the cursor to a position and returns the closest key/value pair.
 func (c *Cursor) Seek(seek []byte) (key, value []byte) {
 	// Move cursor to appropriate block and set to buffer.
-	_, v := c.cursor.Seek(seek)
+	k, v := c.cursor.Seek(seek)
+	if v == nil { // get the last block, it might have this time
+		_, v = c.cursor.Last()
+	} else if bytes.Compare(seek, k) == -1 { // the seek key is less than this block, go back one
+		_, v = c.cursor.Prev()
+	}
```
Comment (benbjohnson) [inline tsdb/engine/bz1/bz1.go]: Good catch. :)

**Comment by benbjohnson** [comment]
@pauldix The fix lgtm but I don't like adding to the interfaces -- especially if it's just for testing. Can you expose a `Shard.Engine()` method and then we can type convert it to `*bz1.Engine` to do the `FlushWAL()`?

e.g.

```go
sh.Engine().(*bz1.Engine).FlushWAL()
```

Then we don't have to add the `Flush()` and `FlushWAL()` to the interfaces.

**Comment by otoolep** [comment]
Testing with this fix looks good, I can no longer reproduce #3781

**Comment by pauldix** [comment]
@benbjohnson That test is in the server test, so making that change would actually tie it to bz1, which will almost certainly change later on. But I agree that it's annoying to change the interface just for the sake of a test.

I guess I can either remove that test and thus the `FlushWAL` part of the interface, or I can keep as is. What you think?

**Comment by benbjohnson** [comment]
Can you force the engine type on the Server for that test? I don't think we want the cruft of FlushWAL() in the interface long term.

**Comment by pauldix** [comment]
would be easier to just remove the test since the actual issue is covered
in the new unit test I added to bz1_test

**Comment by benbjohnson** [comment]
That works for me.

**Comment by pauldix** [comment]
@benbjohnson done :)

**Comment by pauldix** [comment]
cherry-picked to 0.9.3

**Pull request merged**